src/main/java/ch/psc/gui/components/signUp/CreateAccount.java

```diff
@@ -38,69 +49,143 @@ public CreateAccount() {
      * Initializes form and creates input field for creating a new account.
      */
     private void initialize() {
-        this.setSpacing(15);
+        this.setSpacing(40); //TODO: make appearance of textfields unified (in login and signup window)
         this.setPadding(new Insets(10, 20, 10, 20));
 
         usernameField.setPromptText(Config.getResourceText("signup.prompt.username"));
+        //Username field should not be empty
+        RequiredFieldValidator requiredUserNameValidator = new RequiredFieldValidator();
+        requiredUserNameValidator.setMessage(Config.getResourceText("signup.errorLabel.usernameRequired"));
+        usernameField.getValidators().add(requiredUserNameValidator);
+        usernameField.focusedProperty().addListener(new ChangeListener<Boolean>() {
+            @Override
+            public void changed(ObservableValue<? extends Boolean> observable, Boolean oldValue, Boolean newValue) {
+                if (!newValue) usernameField.validate(); //wenn kein neuer Value, dann validate
+            }
+        });
+
+
         emailTextField.setPromptText(Config.getResourceText("signup.prompt.email"));
+        //Email address field should not be empty
+        RequiredFieldValidator requiredEmailValidator = new RequiredFieldValidator();
+        requiredEmailValidator.setMessage(Config.getResourceText("signup.errorLabel.emailRequired"));
+        emailTextField.getValidators().add(requiredEmailValidator);
+        //Email address should be valid
+        RegexValidator inputMailValidator = new RegexValidator();
+        inputMailValidator.setMessage(Config.getResourceText("signup.errorLabel.emailNotValid"));
+        inputMailValidator.setRegexPattern("^[a-zA-Z0-9_!#$%&'*+/=?`{|}~^.-]+@[a-zA-Z0-9.-]+$");
+
+        emailTextField.focusedProperty().addListener(new ChangeListener<Boolean>() {
+            @Override
+            public void changed(ObservableValue<? extends Boolean> observable, Boolean oldValue, Boolean newValue) {
+                if (!newValue) emailTextField.validate(); //wenn kein neuer Value, dann validate
+                emailTextField.getValidators().add(inputMailValidator);
+            }
+        });
+
 
         passwordTextField.setPromptText(Config.getResourceText("signup.prompt.enterPassword"));
+        RegexValidator passwordValidator = new RegexValidator();
+        //Regex, password must contain at least one digit from 0-9, spaces are not allowed, at least 8 characters and at most 20
+        passwordValidator.setRegexPattern("^(?=.*[0-9])"+"(?=\\S+$).{8,20}$");
+        passwordValidator.setMessage(Config.getResourceText("signup.errorLabel.passwordNotValid"));
```

Comment:
Message bitte direkt in den entsprechende Konstruktor einfügen. Bitte gleich bei allen Validator

Reply:
geht leider nicht bei allen Validators - funktioniert nicht bei RequiredFieldValidator.

Reply:
Vorschlag: Ich übernehme die Implementierung des RequiredFieldValidator, füge ihn zum package validator.und ergänze den Konstruktur manuell.